include/pops/treatments.hpp

```diff
@@ -175,29 +177,43 @@ class SimpleTreatment : public BaseTreatment<IntegerRaster, FloatRaster>
         IntegerRaster& infected,
         std::vector<IntegerRaster>& exposed,
         IntegerRaster& susceptible,
-        IntegerRaster&,
+        IntegerRaster& resistant,
+        IntegerRaster& total_hosts,
         const std::vector<std::vector<int>>& suitable_cells) override
     {
         for (auto indices : suitable_cells) {
             int i = indices[0];
             int j = indices[1];
+            int new_exposed_total = 0;
+            int new_infected = 0;
+            int new_susceptible = 0;
+            int new_exposed_individual = 0;
             if (this->application_ == TreatmentApplication::Ratio) {
-                infected(i, j) = infected(i, j) - (infected(i, j) * this->map_(i, j));
+                new_infected = infected(i, j) - (infected(i, j) * this->map_(i, j));
+                infected(i, j) = new_infected;
             }
             else if (this->application_ == TreatmentApplication::AllInfectedInCell) {
-                infected(i, j) = this->map_(i, j) ? 0 : infected(i, j);
+                new_infected = this->map_(i, j) ? 0 : infected(i, j);
+                infected(i, j) = new_infected;
             }
             for (auto& raster : exposed) {
                 if (this->application_ == TreatmentApplication::Ratio) {
-                    raster(i, j) = raster(i, j) - (raster(i, j) * this->map_(i, j));
+                    new_exposed_individual =
+                        raster(i, j) - (raster(i, j) * this->map_(i, j));
+                    raster(i, j) = new_exposed_individual;
+                    new_exposed_total += new_exposed_individual;
                 }
                 else if (
                     this->application_ == TreatmentApplication::AllInfectedInCell) {
                     raster(i, j) = this->map_(i, j) ? 0 : raster(i, j);
+                    new_exposed_total += this->map_(i, j) ? 0 : raster(i, j);
```

Comment:
Can you please do one more change and do this through assigning new_exposed_individual? I just spend some time doubting whether this is even correct. It is correct because `raster(i, j)` is either the original value or 0, but it would be wrong if it would be, e.g., multiplied. Anyway, for clarity and maybe speed, it still is worth to change I think.

Reply:
good catch I missed that in the else if statement.